funnel_test.go

```diff
@@ -271,3 +272,49 @@ func TestOpInProgress(t *testing.T) {
 		t.Error("Expected op to be in progress")
 	}
 }
+
+var getRandomInt = func() (interface{}, error) {
+	time.Sleep(time.Millisecond * 50)
+	res := rand.Int()
+	return &res, nil
+}
+
+/*
+Test ExecuteAndCopyResult function. Validate that ExecuteAndCopyResult returns copied object and not a shared object between the requesting callers.
+*/
+func TestExecuteAndCopyResult(t *testing.T) {
+	fnl := New()
+	var wg sync.WaitGroup
+
+	var num1, num2 *int
+
+	wg.Add(2)
+
+	// preform getRandomInt twice, with same operation id - both goroutines are expected to get same result.
+	// each goroutine stores the returned result, the results would be compared later
+	go func() {
+		defer wg.Done()
+		res, _ := fnl.ExecuteAndCopyResult("opId", getRandomInt)
+		num1 = res.(*int)
+	}()
+
+	go func() {
+		defer wg.Done()
+		res, _ := fnl.ExecuteAndCopyResult("opId", getRandomInt)
+		num2 = res.(*int)
+	}()
+
+	// wait until both goroutines finish
+	wg.Wait()
+
+	// we are expecting ExecuteAndCopyResult to preform deep copy, the returned results for same operation
+	// should have same values but different addresses
+	if *num1 != *num2 {
+		t.Error("Objects' values are expected to be the same. values received:", *num1, ",", *num2)
+	}
+
+	if &num1 == &num2 {
```

Comment:
if num1 and num1 are pointers the correct way to compare them is 
Pointer values can be compared for equality but not for ordering. Two pointer values are considered equal if they point to the same value in memory (or if they are nil). For instance, in the following snippet &pair is equal to ptr2 while &pair is not equal ptr.
func main() {
    pair := [2]int {4, 2}
    ptr := &[2]int {4, 2}
    ptr2 := &pair
 
    if &pair != ptr {
        fmt.Println("pointing different")
    }
    if &pair == ptr2 {
        fmt.Println("pointing the same")
    }
}
Keep in mind that a pointer to a type is not the same as the type itself. Thus Trying to compare the two will cause a type mismatched compilation error.
https://medium.com/learning-the-go-programming-language/comparing-values-in-go-8f7b002e767a
meaning num1==num2 assert WDYT?

Reply:
thanks! 
I'm gonna use assertFalse here, since assert NotEqual wont work for this one :(

```
// Pointer variable equality is determined based on the equality of the
// referenced values (as opposed to the memory addresses).
func NotEqual(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) bool {
```